MagmaSafe.UseCases/Secret/GetSecretUseCase.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading.Tasks;
+using MagmaSafe.Borders.UseCases;
+using MagmaSafe.Borders.Repositories;
+using MagmaSafe.Borders.Shared;
+using MagmaSafe.Shared.Models;
+
+namespace MagmaSafe.UseCases.Secret
+{
+    public class GetSecretUseCase : IGetSecretUseCase
+    {
+        private readonly ISecretRepository secretRepository;
+
+        public GetSecretUseCase(ISecretRepository secretRepository)
+        {
+            this.secretRepository = secretRepository;
+        }
+
+        public async Task<UseCaseResponse<Borders.Entities.Secret>> Execute(string id)
+        {
+            var response = new UseCaseResponse<Borders.Entities.Secret>();
+
+            try
+            {
+                var secret = await secretRepository.GetById(id);
+
+                if (secret != null)
+                {
+                    return response.SetSuccess(secret);
+                }
+                else
+                {
+                    return response.SetNotFound(new ErrorMessage("00.01", $"Unable to find user with id = {id}"));
```

Comment:
Novamente: Não pode ter falando sobre **User**. Isso é um UseCase do Secret.

```suggestion
                    return response.SetNotFound(new ErrorMessage("00.01", $"Unable to find secret with id = {id}"));
```